src/main/java/seedu/address/logic/commands/edit/EditExamCommand.java

```diff
@@ -0,0 +1,227 @@
+import seedu.address.model.module.Module;
+import seedu.address.model.module.ModuleCode;
+import seedu.address.model.module.Timeslot;
+import seedu.address.model.module.Venue;
+import seedu.address.model.module.exam.Exam;
+import seedu.address.model.module.exam.ExamName;
+
+public class EditExamCommand extends EditCommand {
+    public static final String MESSAGE_DUPLICATE_EXAM = "This exam already exists.";
+    public static final String MESSAGE_EDIT_EXAM_SUCCESS = "Edited Exam %s in Module %s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits an exam in the Mod book. "
+            + "\nParameters: "
+            + PREFIX_CODE + "MOD_CODE "
+            + PREFIX_NAME + "EXAM_NAME "
+            + PREFIX_DAY + "DAY "
+            + PREFIX_START + "START_TIME "
+            + PREFIX_END + "END_TIME "
+            + PREFIX_LINK + "LINK "
+            + PREFIX_VENUE + "VENUE "
+            + "\nExample: " + COMMAND_WORD + " exam "
+            + PREFIX_CODE + "CS2103 "
+            + PREFIX_NAME + "Final "
+            + PREFIX_DAY + "02/02/1999 "
+            + PREFIX_START + "10:00 "
+            + PREFIX_END + "11:00 "
+            + PREFIX_LINK + "https://www.youtube.com/watch?v=8mL3L9hN2l4 "
+            + PREFIX_VENUE + "Field";
+    private final Index targetIndex;
+    private final ModuleCode targetModuleCode;
+    private final EditExamDescriptor editExamDescriptor;
+
+    /**
+     * Creates an EditExamCommand to edit the Exam at specified {@code Index} of the specified {@code Module}.
+     */
+    public EditExamCommand(Index targetIndex, ModuleCode targetModuleCode, EditExamDescriptor editExamDescriptor) {
+        requireAllNonNull(targetIndex, targetModuleCode, editExamDescriptor);
+
+        this.targetIndex = targetIndex;
+        this.targetModuleCode = targetModuleCode;
+        this.editExamDescriptor = new EditExamDescriptor(editExamDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        Module module = model.getModule(targetModuleCode);
+        List<Exam> exams = module.getExams();
+
+        if (targetIndex.getZeroBased() >= exams.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_EXAM_DISPLAYED_INDEX);
+        }
+
+        Exam examToEdit = exams.get(targetIndex.getZeroBased());
+        Exam editedExam = createEditedExam(examToEdit, editExamDescriptor);
+
+        if (!examToEdit.isSameExam(editedExam) && model.moduleHasExam(module, editedExam)) {
+            throw new CommandException(MESSAGE_DUPLICATE_EXAM);
+        }
+
+        model.setExam(module, examToEdit, editedExam);
+        model.updateFilteredModuleList(PREDICATE_SHOW_ALL_MODULES);
+        return new CommandResult(String.format(MESSAGE_EDIT_EXAM_SUCCESS, editedExam, module));
```

Comment:
wondering if editing a lesson / exam should make modbook stay in `GuiState.DETAILS`. this means updating the `FilteredModuleList` wouldn't be necessary